src/ui/component/stepper/stepper.tsx

```diff
@@ -2,13 +2,14 @@ import { TransitionGroup, CSSTransition } from 'react-transition-group'
 import { type FC } from 'react'
 import type { Optional } from '@/util/type'
 import { StepperProgress } from './stepperProgress/stepperProgress'
-import type { Step, UseStepper } from './useStepper'
-import { useStepper } from './useStepper'
+import type { Step, UseStepper } from '../../hook/useStepper'
+import { useStepper } from '@/ui/hook/useStepper'
 import { StepperActions } from './stepperActions/stepperActions'
 import './stepper.scss'
 
 export type StepElement = Optional<Omit<Step, 'order'>, 'status'> & {
   content: JSX.Element
+  onValidate?: () => boolean
```

Comment:
`onValidate` is more of an action explicitly called for validation rather than a handler for a specific event, so I'd have called it `validate`.